src/app/course/challenge/challenge-row/challenge-row.component.ts

```diff
@@ -30,4 +43,20 @@ export class ChallengeRowComponent implements OnInit {
     getChallengeType(): string {
         return startCase(this.event.challenge_type?.toLowerCase())
     }
+
+    showDialog(content: PolymorpheusContent<TuiDialogContext>): void {
+        this.dialogService.open(
+            content,
+            {label: 'You are about to delete this challenge!', size: 'l', closeable: true}
+        ).subscribe()
+    }
+
+    deleteChallenge(): void {
+        this.courseEventService.deleteCourseEvent(this.event.id).subscribe( () => {
+            this.notificationsService.show('Event has been successfully deleted.', {
```

Comment:
We no longer use the word `Event` in the user-facing texts. Please use `Assessment` or `Challenge`.